### ISSUE TITLE
Custom Entry subcollections not working

### ISSUE THREAD

**Issue by mchsk** [firecmsco/firecms#504]
Hi!
Just exploring the custom entry from firecms that could be found on demo deployment here https://demo.firecms.co/additional.
I tried adding `locales` subcollection, just like you'd use in the other collections - e.g. `products`.

Fundamentally, I think if the cms can support localizations via subcollections, you would want to allow for this even in the custom view.

So what works:
- the subcollection name appears at the header of the view

What does not:
- instead of collection view where you'd expect adding items with ids such as `en` `de` etc.. there is an error saying: 
`ERROR: No collection found in path `collection_goes_here`. Make sure you have defined a collection for this path in the root navigation.`
- Instead of collection name `collection_goes_here` there is real collection, but (!) the collection is taken from `sideEntityController.open(....path...)` or the `path` of the collection where the custom entity is being referenced. That said, the path from the subcollection seems not to be taken into account at all - it is not part of the error message.

PS. I even created the subcollection manually in firebase, it still isn't being picked up, it seems it had no effect on the issue.

A side-thought: Is it necessary to add "entry" as a suffix at the end of the custom entity title in the custom view? 😸 🤷‍♂️

**Comment by fgatti675** [comment]
Thank you @mchsk 
Could you provide some code reproducing the issue?

**Comment by mchsk** [comment]
Yea why not @fgatti675 

Let's use `SampleConfigurationView.tsx` as an example. The sample idea is that we have a configuration where we want to provide hypothetical landing page configuration where we allow user to set logo, and title. Title is subject to translation.

```
import { Link } from "@mui/icons-material";
import {
    Box,
    Button,
    Card,
    CardActions,
    CardContent,
    Container,
    Grid,
    IconButton,
    Tooltip,
    Typography
} from "@mui/material";

import {
    PropertiesOrBuilders,
    buildCollection,
    useAuthController,
    useSideEntityController
} from "firecms";

const locales = {
    "en-US": "English (United States)",
    "es-ES": "Spanish (Spain)",
    "de-DE": "German"
};

export function SampleConfigurationView() {

    // hook to open the side dialog that shows the entity forms
    const sideEntityController = useSideEntityController();

    // hook to do operations related to authentication
    const authController = useAuthController();

    // sample configuration model
    type SampleConfigurationModel = {
        title: string;
        logo: string;
    };

    // a translation model. it is a subset of the original model.
    // i want to make sure the translation-relevant fields match original model.
    // will only translate the "title" property

    type SampleConfigurationTranslationModel = Partial<SampleConfigurationModel>;

    const sampleConfigurationProperties = {
        logo: {
            name: "Logo",
            description: "Logo",
            dataType: "string",
            storage: {
                storagePath: "images",
                acceptedFiles: ["image/*"],
                metadata: {
                    cacheControl: "max-age=1000000"
                }
            }
        },
        title: {
            name: "Title",
            description: "Title (Max 15 chars)",
            validation: { required: true, max: 15 },
            dataType: "string"
        },
    } as PropertiesOrBuilders<SampleConfigurationModel>;

    const sampleConfigurationTranslationProperties = {
        // Making sure translated title has same configuration as the main one
        title: sampleConfigurationProperties.title
    } as PropertiesOrBuilders<SampleConfigurationTranslationModel>;


    const sampleConfigurationTranslationCollection = buildCollection<SampleConfigurationTranslationModel>({
        path: "translation",
        customId: locales,
        name: "Translation",
        singularName: "Translation",
        properties: sampleConfigurationTranslationProperties
    });

    const sampleConfigurationCollection = buildCollection<SampleConfigurationModel>({
        path: "configuration",
        name: "App Onboarding Settings",
        properties: sampleConfigurationProperties,
        subcollections: [sampleConfigurationTranslationCollection]
    });

    return (
        <Box
            display="flex"
            width={"100%"}
            height={"100%"}>

            <Box m="auto"
                 display="flex"
                 flexDirection={"column"}
                 alignItems={"center"}
                 justifyItems={"center"}>

                <Container maxWidth={"md"}
                           sx={{
                               my: 4
                           }}>

                    <Grid container rowSpacing={5} columnSpacing={2}>

                        <Grid item xs={12}>
                            <Typography variant={"h4"}>
                                Custom page
                            </Typography>
                            <Typography>
                                {authController.user
                                    ? <>Logged in
                                        as {authController.user.displayName}</>
                                    : <>You are not logged in</>}
                            </Typography>
                        </Grid>

                        <Grid item xs={12} sm={4}>
                            <Card variant="outlined" sx={{ height: "100%" }}>
                                <CardContent>
                                    <Typography>
                                        Here you can configure title and logo.
                                    </Typography>
                                </CardContent>

                                <CardActions>
                                    <Button
                                        onClick={() => sideEntityController.open({
                                            // let's say we're configuring some landing page with title and logo
                                            entityId: "landingPageConfiguration",
                                            path: "configuration",
                                            collection: sampleConfigurationCollection,
                                            width: 800
                                        })}
                                        color="primary">
                                        Open
                                    </Button>
                                </CardActions>
                            </Card>
                        </Grid>
                    </Grid>
                </Container>
            </Box>
        </Box>
    );
}
```

In the `App.tsx` you'd use this array (of single obj) of custom views while instantiating the `FirebaseCMSApp`:
```
const customViews: CMSView[] = [{
        path: "configuration",
        name: "Sample Configuration",
        group: "Main",
        description: "Sample Configuration",
        view: <SampleConfigurationView/>
    }];
```

**Comment by mchsk** [comment]
Result:

1) The page looks good:
<img width="925" alt="Screenshot 2023-06-02 at 21 07 52" src="https://github.com/firecmsco/firecms/assets/8226069/557a5941-4023-4d89-8085-01c39c292a42">

2) The entry (please not my not about the "entry" suffix 😅) looks good too
<img width="975" alt="Screenshot 2023-06-02 at 21 08 04" src="https://github.com/firecmsco/firecms/assets/8226069/5ba8e514-835f-42a8-a8e4-695f6576ba93">

3) Once you click on "Translation" tab, you see this:
<img width="1403" alt="Screenshot 2023-06-02 at 21 08 16" src="https://github.com/firecmsco/firecms/assets/8226069/8a40afba-b123-4c98-8099-45a71b763acd">

**Comment by fgatti675** [comment]
fixed in version 2.0.0 :)

**Comment by fgatti675** [comment]
Thanks! cleaning up for next version
